src/Mailjet/Client.php

```diff
@@ -28,21 +28,25 @@ class Client
     private $apikey;
     private $apisecret;
 
-    private $version = 'v3/';
-    private $secure = true;
+    private $version = Config::MAIN_VERSION;
+    private $url = Config::MAIN_URL;
+    private $secured = Config::SECURED;
     private $call = true;
+    private $settings = [];
 
     /**
      * Client constructor requires:
      * @param string  $key    Mailjet API Key
      * @param string  $secret Mailjet API Secret
      * @param boolean $call   performs the call or not
      */
-    public function __construct($key, $secret, $call = true)
-    {
-        $this->call = $call;
+    public function __construct($key, $secret, array $settings = [])
+    {           
         $this->apikey = $key;
         $this->apisecret = $secret;
+        $this->initSettings($settings);
+        $this->settings["changed"] = true;
+        $this->setSettings();
```

Comment:
I get that, but I don't understand why you're calling both consequently (one immediately after the other)